File: scraper/scrape.py
#!/usr/bin/python

import sys
import json
import sqlite3
import re
from datetime import datetime

with open(sys.argv[1]) as file:
	data=json.load(file)
	data=data['array']
	dbcon = sqlite3.connect(sys.argv[2])
	dbcon.isolation_level = None
	cursor = dbcon.cursor()
	cursor.execute('begin')
	regex = re.compile(".*#(.*)\s+\((.*)\)")
	for d in data:
		addr=d['addr']
		r = regex.search(addr)
		if r == None:
			grp= ['','']
		else:
			grp=r.groups()
		cursor.execute('''INSERT OR REPLACE INTO `distributori`(`id`, `name`, `bnd`, `lat`, `lon`, `addr`, `comune`, `provincia`) 
                  VALUES (:id,:name,:bnd,:lat,:lon,:addr,:comune,:provincia)''',
                  {'id':d['id'],'name':d['name'],'bnd':d['bnd'],'lat':d['lat'],'lon':d['lon'],'addr':d['addr'],'comune':grp[0].title(),'provincia':grp[1]})
                
                carb=d['carburanti']
                
                for c in carb:
 			cursor.execute('''INSERT INTO `prezzi`(`id_d`, `dIns`, `carb`, `isSelf`, `prezzo`, `dScrape`)  
                	  VALUES (:id_d,:dIns,:carb,:isSelf,:prezzo,:dScrape)''',
                	  {'id_d':d['id'],'dIns':d['dIns'],'carb':c['carb'],'isSelf':c['isSelf'],'prezzo':c['prezzo'],'dScrape':datetime.strftime(datetime.today(),'%Y%m%d')})               	

	cursor.execute('end')
	dbcon.commit()
	dbcon.close()

File: scraper/scrape.php
<?php
if(php_sapi_name()=="cli")
{
    $regione=$argv[1];
}

$json=json_decode(file_get_contents($regione),true);
$arr=$json['array'];

echo "Processing ".$regione."\n";

try{
    $file_db = new PDO('sqlite:db/scrape.sqlite');
    $file_db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    $dis_sql='INSERT OR REPLACE INTO `distributori`(`id`, `name`, `bnd`, `lat`, `lon`, `addr`, `comune`, `provincia`) ';
    $dis_sql.='VALUES (:id,:name,:bnd,:lat,:lon,:addr,:comune,:provincia)';
    $dis_sth   = $file_db->prepare($dis_sql);

    $carb_sql='INSERT OR REPLACE INTO `prezzi`(`id_d`, `dIns`, `carb`, `isSelf`, `prezzo`) ';
    $carb_sql.='VALUES (:id_d,:dIns,:carb,:isSelf,:prezzo)';
    $carb_sth   = $file_db->prepare($carb_sql);

    $file_db->beginTransaction();
    foreach($arr as $entry){
	$addr=$entry['addr'];
	$r=preg_match('/.*#(.*)\s+\((.*)\)/',$addr, $matches);
	if($r==0||$r===NULL) $matches=['','',''];
	$dis_sth->execute(array(':id' => $entry['id'], ':name' => $entry['name'],':bnd' => $entry['bnd'],':lat' => $entry['lat'],':lon' => $entry['lon'],':addr' => $entry['addr'],':comune' => ucwords(strtolower(trim($matches[1]))),':provincia' => $matches[2]) );

        $carburanti=$entry['carburanti'];
        foreach ($carburanti as $carb){
            $carb_sth->execute(array(':id_d' => $entry['id'], ':dIns' => $entry['dIns'],':carb' => $carb['carb'],':isSelf' => $carb['isSelf'],':prezzo' => $carb['prezzo']) );
        }
    }
    $file_db->commit();

    $file_db = null;
    $memory_db = null;
  }
  catch(PDOException $e) {
    echo $e->getMessage();
  }

?>
